### PR TITLE
MULE-19092: add interceptor test

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

File: extensions-xml-support/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java [hunk 1]
```diff
@@ -607,10 +628,28 @@ public ProcessorInterceptor get() {
...
+    List<ProcessorInterceptor> processorInterceptors;
+
+    public AfterWithCallbackInterceptor(List<ProcessorInterceptor> processorInterceptors) {
+      this.processorInterceptors = processorInterceptors;
+    }
+
     @Override
     public void after(ComponentLocation location, InterceptionEvent event, Optional<Throwable> thrown) {
+      processorInterceptors.add(this);
+      this.afterClassLoader = Thread.currentThread().getContextClassLoader();
```
Comment (elrodro83) [inline extensions-xml-support/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java:642]: static import

File: extensions-xml-support/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java [hunk 2]
```diff
@@ -125,6 +130,16 @@ public void scOperation() throws Exception {
     assertThat(setPayloadOperation.getParameters().get("encoding").resolveValue(), is("UTF-8"));
   }
 
+  @Test
+  public void interceptionClassLoader() throws Exception {
```
Comment (elrodro83) [inline extensions-xml-support/src/test/java/org/mule/test/integration/interception/ProcessorInterceptorFactoryTestCase.java:134]: `@Issue`

**Review by elrodro83 — APPROVED** [review]

**Comment by jcgrethe** [comment]
--validate

**Comment by jcgrethe** [comment]
--validate

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**